### PR TITLE
deploy: add annotation for disabling DVO check (PROJQUAY-3680)

### PR DESCRIPTION
**Pull request by Marcusk19**
disable min replicas check for staging environment where we only have a single replica

### CODE REVIEW

File: deploy/openshift/quay-py3-app.yaml [hunk 1]
```diff
@@ -148,6 +148,8 @@ objects:
           ${{QUAY_APP_COMPONENT_LABEL_KEY}}: ${{QUAY_APP_COMPONENT_LABEL_VALUE}}
         annotations:
           ${{QUAY_APP_COMPONENT_ANNOTATIONS_KEY}}: ${{QUAY_APP_COMPONENT_ANNOTATIONS_VALUE}}
+          ignore-check.kube-linter.io/minimum-replicas: ${{QUAY_DISABLE_MIN_THREE_REPLICAS_CHECK}}
```
Comment (syed) [inline deploy/openshift/quay-py3-app.yaml:151]: can you rename to `QUAY_DISABLE_MIN_REPLICAS_CHECK`

File: deploy/openshift/quay-py3-app.yaml [hunk 2]
```diff
@@ -393,6 +395,9 @@ parameters:
   - name: QUAY_APP_READINESS_PROBE_TIMEOUT_SECONDS
     value: "10"
     displayName: quay app readiness probe timeout
+  - name: QUAY_DISABLE_MIN_THREE_REPLICAS_CHECK
+    value: ""
```
Comment (syed) [inline deploy/openshift/quay-py3-app.yaml:399]: one thing I'm not sure is if having an empty value keeps the check enabled. We need to validate this in stage before we deploy in prod.

**Review by syed — CHANGES_REQUESTED** [review]

File: deploy/openshift/quay-py3-app.yaml (hunk 2 above)
Comment (syed) [reply deploy/openshift/quay-py3-app.yaml:399]: or we can even evaluate this in our dev cluster

Comment (Marcusk19) [reply deploy/openshift/quay-py3-app.yaml:399]: @syed sounds good to me, fwiw [the doc](https://github.com/stackrox/kube-linter/blob/main/docs/configuring-kubelinter.md#ignoring-violations-for-specific-cases) I was looking at suggests that an empty value will still disable the check

Comment (syed) [reply deploy/openshift/quay-py3-app.yaml:399]: ok that might be an issue here. If that's the case, can you also make a param of the annotation key as well? `ignore-check.kube-linter.io/minimum-replicas` will become a param with default empty value

Comment (Marcusk19) [reply deploy/openshift/quay-py3-app.yaml:399]: done 🙂

**Comment by syed** [comment]
can you also add this to `quay-py3-cloudflare-app.yaml` file

**Comment by Marcusk19** [comment]
it's in `quay-py3-cloudflare-app.yaml` too

**Review by syed — APPROVED** [review]
LGTM

**Pull request merged**